samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,164 @@
+.. _coremark_sample:
+
+CoreMark
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The sample demonstrates how to run the `CoreMark®`_ benchmark to evaluate the performance of a core.
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+Overview
+********
+
+The sample runs the CoreMark benchmark on the target CPU.
+CoreMark evaluates the CPU efficiency by performing different algorithms, such as state machine, CRC calculation, matrix manipulation, and list processing (find and sort).
+
+To run the CoreMark benchmark on the preferred core, press the corresponding button.
+For the button assignment, see the :ref:`coremark_user_interface` section.
+When the benchmark has completed, you can press the same button to restart it.
+If you want to run the sample upon startup, enable the :kconfig:option:`CONFIG_COREMARK_MODE_FLASH_AND_RUN` Kconfig option.
```

Comment:
`CONFIG_APP_MODE_FLASH_AND_RUN`. Btw. IIRC we cannot use this form of Kconfig reference for sample/app specific options